### PR TITLE
Added slab renderer

### PR DESCRIPTION
**Pull request by mrpimpunicorn**
The code to half the size of the block smells, but it works. Could you check if it supports wood/cobblestone slabs? I've only been able to test stone as I don't have a copy of Java 8 or Minecraft 1.7.3 to connect to the server and test with.

### CODE REVIEW

File: TrueCraft.Client/Rendering/Blocks/SlabRenderer.cs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            BlockRenderer.RegisterRenderer(DoubleStoneSlabBlock.BlockID, new SlabRenderer());
+        }
+
+        public enum SlabType : byte
+        {
+            Stone = 0,
+            Sandstone = 1,
+            Wood =2,
+            Cobblestone = 3
+        }
```
Comment (SirCmpwn) [inline TrueCraft.Client/Rendering/Blocks/SlabRenderer.cs]: Type enums go in the class they belong to - in this case, let's rename `StoneSlabBlock` to `SlabBlock`, and make `DoubleStoneSlabBlock` into `DoubleSlabBlock` and have it inherit from `SlabBlock`.

File: TrueCraft.Client/Rendering/Blocks/SlabRenderer.cs [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+                topTexture + Vector2.UnitX + Vector2.UnitY,
+                topTexture + Vector2.UnitY,
+                topTexture,
+                topTexture + Vector2.UnitX,
+                // Negative Y
+                bottomTexture + Vector2.UnitX + Vector2.UnitY,
+                bottomTexture + Vector2.UnitY,
+                bottomTexture,
+                bottomTexture + Vector2.UnitX
+            };
```
Comment (SirCmpwn) [inline TrueCraft.Client/Rendering/Blocks/SlabRenderer.cs]: We need to generate texture arrays once and reuse them for all rendered blocks. See how logs do it, for example. It's very wasteful to allocate a new array for every block rendered.

**Comment by mrpimpunicorn** [comment]
That should do it. Still, test the other slab types since I'm not able to.

File: TrueCraft.Client/Rendering/Blocks/SlabRenderer.cs [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+                case SlabBlock.SlabMaterial.Stone:
+                    return StoneTextureMap;
+                case SlabBlock.SlabMaterial.Standstone:
+                    return SandstoneTextureMap;
+                case SlabBlock.SlabMaterial.Wooden:
+                    return WoodTextureMap;
+                case SlabBlock.SlabMaterial.Cobblestone:
+                    return CobbleTextureMap;
+                default:
+                    return null;
```
Comment (SirCmpwn) [inline TrueCraft.Client/Rendering/Blocks/SlabRenderer.cs]: This will probably lead to a client crash. We should return stone slab textures as the default.

Comment (mrpimpunicorn) [inline TrueCraft.Client/Rendering/Blocks/SlabRenderer.cs]: Done.

**Comment by SirCmpwn** [comment]
Oh, that's actaully not right. The second one from the left should be *cobble*, not *stone*.

**Pull request merged**